### PR TITLE
Add ZigUSB support

### PR DESCRIPTION
**Pull request by xyzroe**
All tests passed

Screenshot 2024-02-15 at 11 18 13 PM

### CODE REVIEW

File: src/devices/xyzroe.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+};
+
+const bindCommandList = {
+    'on/off': 0x00,
+    'toggle': 0x01,
+    'change level up': 0x02,
+    'change level down': 0x03,
+    'change level up with off': 0x04,
+    'change level down with off': 0x05,
+    'recall scene 0': 0x06,
```
Comment (Koenkk) [inline src/devices/xyzroe.ts:28]: Please make these all snake case, e.g. `recall_scene_0`

File: src/devices/xyzroe.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+        getSortedList(bindCommandList)).withEndpoint(epName));
+    return features;
+}
+
+const definitions: Definition[] = [
+    {
+        zigbeeModel: ['ZigUSB'],
+        model: 'ZigUSB',
+        vendor: 'xyzroe',
+        description: '[Zigbee USB power monitor and switch](https://xyzroe.cc/ZigUSB)',
```
Comment (Koenkk) [inline src/devices/xyzroe.ts:244]: Let's not add the link here but in the docs (zigbee2mqtt.io)

File: src/devices/xyzroe.ts (hunk 1 above)
Comment (xyzroe) [reply src/devices/xyzroe.ts:28]: ok, but they are used in the interface, I wanted to make it look visually pleasing.

File: src/devices/xyzroe.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+                payload = {0x4002: {value: data, type: 32 /* uint8 */}};
+                break;
+            }
+            await entity.write('genOnOffSwitchCfg', payload);
+        },
+    } satisfies Tz.Converter,
+    zigusb_on_off_invert: {
+        key: ['state', 'on_time', 'off_wait_time'],
+        convertSet: async (entity, key, value, meta) => {
+            const state = utils.isString(meta.message.state) ? meta.message.state.toLowerCase() : null;
```
[the remark this answers is not in the document]
Comment (xyzroe) [reply src/devices/xyzroe.ts:86]: The difference from the usual on-off is that it is inverted. Еhe device sends OnOff = 1 when off and vice versa. This device is so unusual.

File: src/devices/xyzroe.ts (hunk 2 above)
Comment (xyzroe) [reply src/devices/xyzroe.ts:244]: Ok. No problem, I did it out of old habit)
But I really don’t really understand how to add a link there if the description is auto-generated.

Comment (xyzroe) [reply src/devices/xyzroe.ts:244]: I tried to compile the documentation locally - but my device did not appear in the list, the MD file was not created. (indicated INCLUDE_DEVICE=ZigUSB)
I assume that while the device is not in the zigbee-herdsman-converters, it cannot be added to the documentation? Or I did something wrong.  

I'm not sure that you even need to build locally to add custom description. 
Tell me where to look please

Comment (Koenkk) [reply src/devices/xyzroe.ts:244]: You can just create the `.md` file, only the non generated part has to be added (between `<!-- Notes BEGIN: You can edit here. Add "## Notes" headline if not already present. -->`)

Comment (xyzroe) [reply src/devices/xyzroe.ts:244]: I read it, but description is autogenerated field. And link is located in description. I saw the same in other devices. Don't understand how to add link to description if it's autogenerated...

Comment (Koenkk) [reply src/devices/xyzroe.ts:244]: I meant using `description: 'Zigbee USB power monitor and switch',` and linking to the device in the `## Notes` section (not in the description)

Comment (xyzroe) [reply src/devices/xyzroe.ts:244]: But lot of devices use link exactly in description. Is that option deprecated?

File: src/devices/xyzroe.ts (hunk 1 above)
Comment (xyzroe) [reply src/devices/xyzroe.ts:28]: Screenshot 2024-02-18 at 9 48 36 PM
will snake case look better in the interface? 

just want confirm before commit

File: src/devices/xyzroe.ts (hunk 2 above)
Comment (Koenkk) [reply src/devices/xyzroe.ts:244]: Yes, I prefer not to use it anymore

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**